taipy/core/job/_job_manager.py

```diff
@@ -94,7 +94,7 @@ def _is_deletable(cls, job: Union[Job, JobId]) -> ReasonCollection:
         if isinstance(job, str):
             job = cls._get(job)
 
-        if not job.is_finished():
+        if job and not job.is_finished():
```

Comment:
I don't get this change. 

1. It does not make sense to pass a `None` value as job.

2. It breaks the reason logic. 
    With this change, calling `_is_deletable(None)` will return True with no Reason. So a user 
    could try to delete the job (which is `None`) which will raise an error.

3. For consistency, it would require to change all the APIs to access None values in similar methods

Reply:
Is it possible that cls._get() returns None ?

I made that change because it happened is it better to leave the code as is because it makes sense or protect it ?

Reply:
OK I get your point. It is not just that you pass a None job, it is also that the job_id does not exist.

I believe we should protect from this, indeed, but not in that way. Let's discuss this.

Reply:
as far as I know, every call to `is_deletable()` in `gui_core `is protected and cannot pass `None`

Reply:
Ticket https://github.com/Avaiga/taipy/issues/1805 should handle this fix.